linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -750,37 +751,18 @@ private final class IRChecker(unit: LinkingUnit,
         val qualType = typecheckExpr(qualifier, env)
         qualType match {
           case ClassType(cls) =>
-            val maybeClass = tryLookupClass(cls)
-            val kind = maybeClass.fold(_.kind, _.kind)
+            val c = lookupClass(cls)
+            val kind = c.kind
             if (!kind.isClass) {
               reportError(s"Cannot select $item of non-class $cls")
             } else {
-              /* Actually checking the field is done only if
-               *
-               * * the class exists, and
-               * * it has instances (including instances of subclasses).
-               *
-               * The latter test is necessary because the BaseLinker can
```

Comment:
This part of the comment should be preserved.

Reply:
Done.